io/io/src/TFile.cxx

```diff
@@ -851,7 +851,7 @@ void TFile::Init(Bool_t create)
       }
    }
 
-   {
+   if (addToList){
       R__LOCKGUARD(gROOTMutex);
       gROOT->GetListOfFiles()->Add(this);
       gROOT->GetUUIDs()->AddUUID(fUUID,this);
```

Comment:
One would need to study this, but the proper functioning of TRef may requires that this insert is done always ....

Reply:
Do you mean in the list of files or UUIDs? Note that right now, the registration is on by default, i.e. no change but for TBufferMergerFile.